spec/ssh_spec.rb

```diff
@@ -120,7 +134,70 @@
       ).to receive(
         :error,
       ).with(
-        /mock_ssh_command.*mock_jump_user@mock_jump_host.*rossi@mock_host/,
+        /SSH returned error while connecting to #{tt_ssh.ssh_target}.*mock_ssh_command.*mock_jump_user@mock_jump_host.*rossi@mock_host/m,
+      )
+      expect { tt_ssh.error! }.to raise_error(
+        TasteTester::Exceptions::SshError,
+      )
+    end
+    after do
+      TasteTester::Config.restore(config_hash)
+    end
+  end
+
+  context 'test custom ssh configs linux - generated ssh command' do
+
+    let(:mock_generated_cmd) do
+      'mock_ssh_command_bin -o option1 -o option2 mock_user@mock_host'
```

Comment:
your template does --user but this does mock_user@ - how does this tests pass?

Reply:
This is mocking the Mixlib::ShellOut response which takes in the "generator command" and returns a vanilla ssh command. It could be a random vanilla ssh command because the generator command could return whatever.

Reply:
oh I see. OK yeah, that makes sense.